### PR TITLE
Functionality to add a line item property

### PR DESCRIPTION
**Pull request by jnava19**
Functionality can be reviewed at: https://twigs-test.myshopify.com/ 

This extension was created before to display when a product is a FINAL SALE, conditions is compare_at_price > 50%

https://screenshot.click/10-48-z6va1-27emn.png

This new feature will also add a line item property to qualifying line items so that they can be returned through the return app.

https://screenshot.click/10-45-bnvoo-m61pf.png

**Code to be reviewed src/extension.jsx**

Thank you!

### CODE REVIEW

File: extensions/discounts-view/dist/discounts-view.js [hunk 1]
```diff
@@ -17516,21 +17510,27 @@ ${errorInfo.componentStack}`);
...
+    const api = (0, import_react8.useContext)(ExtensionApiContext);
     if (api == null) {
       throw new CheckoutUIExtensionError("You can only call this hook when running as a UI extension.");
     }
     return api;
   }
 
   // node_modules/@shopify/ui-extensions-react/build/esm/surfaces/checkout/hooks/subscription.mjs
-  var import_react10 = __toESM(require_react(), 1);
+  var import_react9 = __toESM(require_react(), 1);
```
Comment (siakaramalegos) [inline extensions/discounts-view/dist/discounts-view.js:17530]: Why do you have both react8 and react9? Are those React version numbers? Ditto with all the other numbered variables. It makes it harder to understand what's going on.

File: extensions/discounts-view/dist/discounts-view.js [hunk 2]
```diff
@@ -17564,29 +17579,48 @@ ${errorInfo.componentStack}`);
   }
 
   // extensions/discounts-view/src/extension.jsx
-  var import_react11 = __toESM(require_react());
+  var import_react10 = __toESM(require_react());
```
Comment (siakaramalegos) [inline extensions/discounts-view/dist/discounts-view.js:17582]: Did the original code really mix `var` and `const`? Such a weird mix of different strategies

Comment (jnava19) [reply extensions/discounts-view/dist/discounts-view.js:17582]: I can't really comment on the dist files, they are generated by Shopify CLI and I didn't do. My code is at  src/extension.jsx

File: extensions/discounts-view/dist/discounts-view.js [hunk 3]
```diff
@@ -17564,29 +17579,48 @@ ${errorInfo.componentStack}`);
...
   var import_jsx_runtime4 = __toESM(require_jsx_runtime());
   function Extension() {
     const { query } = useApi();
-    const { cost, merchandise, quantity } = useTarget();
-    const [compareAtPrices, setCompareAtPrices] = (0, import_react11.useState)({});
-    console.log(merchandise.id);
-    (0, import_react11.useEffect)(() => {
+    const { id, cost, merchandise, quantity, attributes } = useTarget();
+    const applyCartLinesChange = useApplyCartLinesChange();
+    const [compareAtPrices, setCompareAtPrices] = (0, import_react10.useState)({});
```
[the remark this answers is not in the document]
Comment (jnava19) [reply extensions/discounts-view/dist/discounts-view.js:17588]: Good point, but again, I only write the code for the jsx, the compiler is taking care of the rest, I can definitely bring those points to discussion.

**Review by siakaramalegos — COMMENTED** [review]
I'm not really sure I can review this well since I don't know much about the original codebase. Plus I'm struggling understanding what's going on with the numbered variables.

**Comment by jnava19** [comment]
> I'm not really sure I can review this well since I don't know much about the original codebase. Plus I'm struggling understanding what's going on with the numbered variables.

True, that's why I mentioned to review only the source code at /src/extension.jsx
extensions/discounts-view/src/extension.jsx

**Comment by jnava19** [comment]
Will close and open a new one to implement on a different approach to avoid https://github.com/Shopify/checkout-web/issues/28651

**Pull request closed without merging**